docs/addon-guide.md

```diff
@@ -0,0 +1,193 @@
+Addons
+
+An addon is an extension of the Hub and consists of a container (image) and a
+manifest (Custom Resource). The CR defines the addon and registers it with the Hub.
+It is intended to provide business logic that contributes to the application 
+*knowledge-base* (inventory) or produces modernization assets. An addon is essentially
+just a binary written in any language that is integrated through the Hub API. The addon
+must report its status using the task (addon) report:
+1. Report addon started: `POST /tasks/:id/report` Task.State=Running.
+2. Report addon activity & progress: `PUT /tasks/:id/report` (repeat as needed).
+3. Report addon completed: `PUT /tasks/:id/report` Task.State=(Succeeded|Failed)
+
+The knowledge about an application consists of:
+- Application (model).
+- Source repository.
+- Tags.
+- Application Facts (key/value pair).
+- Files (stored in the application *bucket*).
+- Assessments.
+
+Modernization assets may be stored in:
+- Source repository.
+- Application Facts (key/value pair).
+- Files (stored in the application *bucket*).
+
+## Definition
+
+An Addon is defined and registered with the Hub using a Custom Resource (CR).
+
+* **image** - The addon image.
+* **resources** - An *(optional)* standard k8s pod container [resources](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers) specification.
+* **mounts** - An optional list of volumes to be mounted in the addon pod.
+    * **claim** - PVC name.
+    * **name** - The name of the directory in `/mnt` to mount the volume.
```

Comment:
Good suggestion.
There really isn't a use case for random mounts and primarily supported the shared maven PVC. This is replaced by a generic RWX PVC that will be mounted in all task pods as part of the optional-RWX solution that should merge in the next few weeks.  That PR will includes changes to this document that will include the mount location.
I'd prefer not to invest in explaining the _mounts_ further in this PR since it will be short lived if you don't mind.